command/agent/cache/cacheboltdb/bolt.go

```diff
@@ -0,0 +1,344 @@
+package cacheboltdb
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hashicorp/go-hclog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/hashicorp/go-multierror"
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	// Keep track of schema version for future migrations
+	storageVersionKey = "version"
+	storageVersion    = "1"
+
+	// DatabaseFileName - filename for the persistent cache file
+	DatabaseFileName = "vault-agent-cache.db"
+
+	// rootBucketName - naming the root bucket
+	rootBucketName = "root"
+
+	// SecretLeaseType - Bucket/type for leases with secret info
+	SecretLeaseType = "secret-lease"
+
+	// AuthLeaseType - Bucket/type for leases with auth info
+	AuthLeaseType = "auth-lease"
+
+	// TokenType - Bucket/type for auto-auth tokens
+	TokenType = "token"
```

Comment:
Looks like this bucket is carved out for the auto-auth token, but can that be stored in the root directly?

Reply:
Perhaps? Though I left it as a separate bucket since this is how the cachememdb behaves, namely old auto-auth tokens are not removed if a new one comes in, and they aren't expired. So I added this bucket out of abundance of caution, in case a lease that needed to be restored relied on a previous auto-auth token.